app/models/export_format.rb

```diff
@@ -0,0 +1,99 @@
+# License: AGPL-3.0-or-later WITH Web-Template-Output-Additional-Permission-3.0-or-later
+class ExportFormat < ActiveRecord::Base
+  # name - string that refers to the name of the nonprofit
+  # date_format - a string that refers to the desired date format
+  # show_currency - boolean that decides whether the currency should be displayed or not
+  # custom_columns_and_values - customizes values and columns from the export
+
+  belongs_to :nonprofit
+
+  validates :name, presence: true
+  validates :nonprofit_id, presence: true
+
+  validates_with PosgresqlDateFormatValidator, { attribute_name: :date_format }
+
+  validate :valid_custom_columns_and_values?
+
+  after_validation do
+    normalize_custom_columns
+  end
```

Comment:
A couple things:

* Was there a particular reason you did this as an `after_validation` step? I think it'd make sense to do it before_validation since the goal here is to get the model into a valid state.
* Generally, it's best to use the single line version of the these callbacks, i.e.: `after_validation :normalize_custom_columns`

Reply:
The reason I decided to do it after the validation is that not being normalized with the double quotes does not make it invalid. The validation on that attribute makes sure that it has the minimal keys and that they are supported, and after that we normalize the customized values to adequate them to the query, so I think it doesn't make sense to apply the normalization on the information before we know if it's valid.

Reply:
I'm not sure I totally agree but it's not a major issue, let's go with it in `after_validation`